cloud/amazon/ec2_asg_facts.py

```diff
@@ -0,0 +1,351 @@
+    description: Name of launch configuration associated with the ASG.
+    returned: success
+    type: str
+    sample: "public-webapp-production-1"
+load_balancer_names:
+    description: List of load balancers names attached to the ASG.
+    returned: success
+    type: list
+    sample: ["elb-webapp-prod"]
+max_size:
+    description: Maximum size of group
+    returned: success
+    type: int
+    sample: 3
+min_size:
+    description: Minimum size of group
+    returned: success
+    type: int
+    sample: 1
+new_instances_protected_from_scale_in:
+    description: Whether or not new instances a protected from automatic scaling in.
+    returned: success
+    type: boolean
+    sample: "false"
+placement_group:
+    description: Placement group into which instances are launched, if any.
+    returned: success
+    type: str
+    sample: None
+status:
+    description: The current state of the group when DeleteAutoScalingGroup is in progress.
+    returned: success
+    type: str
+    sample: None
+tags:
+    description: List of tags for the ASG, and whether or not each tag propagates to instances at launch.
+    returned: success
+    type: list
+    sample: [
+        {
+            "key": "Name",
+            "value": "public-webapp-production-1",
+            "resource_id": "public-webapp-production-1",
+            "resource_type": "auto-scaling-group",
+            "propagate_at_launch": "true"
+        },
+        {
+            "key": "env",
+            "value": "production",
+            "resource_id": "public-webapp-production-1",
+            "resource_type": "auto-scaling-group",
+            "propagate_at_launch": "true"
+        }
+    ]
+termination_policies:
+    description: A list of termination policies for the group.
+    returned: success
+    type: str
+    sample: ["Default"]
+'''
+
+try:
+    import boto3
+    from botocore.exceptions import ClientError
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+def match_asg_tags(tags_to_match, asg):
+    for key, value in tags_to_match.iteritems():
+        for tag in asg['Tags']:
+            if key == tag['Key'] and value == tag['Value']:
+                break
+        else: return False
+    return True
+
+def find_asgs(conn, module, name=None, tags=None):
+    """
+    Args:
+        conn (boto3.AutoScaling.Client): Valid Boto3 ASG client.
+        name (str): Optional name of the ASG you are looking for.
+        tags (dict): Optional dictionary of tags and values to search for.
+
+    Basic Usage:
+        >>> name = 'public-webapp-production'
+        >>> tags = { 'env': 'production' }
+        >>> conn = boto3.client('autoscaling', region_name='us-west-2')
+        >>> results = find_asgs(name, conn)
+
+    Returns:
+        List
+        [
+            {
+                "auto_scaling_group_arn": "arn:aws:autoscaling:us-west-2:275977225706:autoScalingGroup:58abc686-9783-4528-b338-3ad6f1cbbbaf:autoScalingGroupName/public-webapp-production",
```

Comment:
This looks like a real account number - I'd recommend changing it to a dummy like 123456789012